src/net_processing.cpp

```diff
@@ -227,28 +227,28 @@ struct Peer {
     bool m_wtxid_relay{false};
 
     struct TxRelay {
-        mutable RecursiveMutex cs_filter;
-        // We use fRelayTxes for two purposes -
+        mutable RecursiveMutex m_bloom_filter_mutex;
+        // We use m_relay_txs for two purposes -
         // a) it allows us to not relay tx invs before receiving the peer's version message
         // b) the peer may tell us in its version message that we should not relay tx invs
         //    unless it loads a bloom filter.
-        bool fRelayTxes GUARDED_BY(cs_filter){false};
-        std::unique_ptr<CBloomFilter> pfilter PT_GUARDED_BY(cs_filter) GUARDED_BY(cs_filter){nullptr};
+        bool m_relay_txs GUARDED_BY(m_bloom_filter_mutex){false};
```

Comment:
In 2533ea5161d2cb15b3a43e0fea9b146f525aea97

Noting that we have 3 variables named `m_relay_txs`: in `NodeEvictionCandidate`, `Peer.TxRelay`, and `CNodeStateStats`. And we have a `CNode.m_relays_txs` (very similar but not same name) where `CNode.m_relays_txs == Peer.m_tx_relay.m_relay_txs` always, and they are not the same thing as `Peer.m_tx_relay != nullptr`.

Reply:
Yes, this is slightly unfortunate. Please let me know if you have suggestions for improving the naming.

Reply:
Is there a reason for the bool to exist? What would happen if we delayed initialization of the structure to when the version message is received?

Something similar is done for address relay, IIRC.

Reply:
@MarcoFalke yes, that's the plan. This PR is intended to be a simple refactor, and delaying initialization of the `TxRelay` struct requires more focused discussion so should be done in a separate PR.

I've opened PR #22778, which implements delaying constructing `m_tx_relay` until the `version` message is received. Note however that `m_relay_txs` is still required - a peer that we've offered `NODE_BLOOM` services to may send us a version with `fRelay=0`, but later request that we start announcing transactions by sending `filterload` or `filterclear`, so we need to initialize the `TxRelay` struct during `version` handling but use `m_relay_txs` to gate whether we announce transactions.

I'm going to mark this conversation as resolved, but please let me know in #22778 if you have any other thoughts/feedback.